pkg/cmd/pulumi/operations/up.go

```diff
@@ -50,8 +51,19 @@ import (
 
 // The default number of parallel resource operations to run at once during an update, if --parallel is unset.
 // See https://github.com/pulumi/pulumi/issues/14989 for context around the cpu * 4 choice.
-var defaultParallel = int32(runtime.NumCPU()) * 4 //nolint:gosec // NumCPU is an int32 internally,
-//                                                                  but the NumCPU function returns an int.
+var osDefaultParallel = int32(runtime.NumCPU()) * 4 //nolint:gosec // NumCPU is an int32 internally,
+// but the NumCPU function returns an int.
+func convertParallel(p int) int32 {
```

Comment:
I think this worked better as a single function as in one of your earlier pushes. You can just call it in the flag setup like:
```
	cmd.PersistentFlags().Int32VarP(
		&parallel, "parallel", "p", defaultParallel(),
		"Allow P resource operations to run in parallel at once (1 for no parallelism).")
```